bandersnatch/point.go

```diff
@@ -216,19 +213,17 @@ func NewPointAffine(x, y fp.Element) PointAffine {
 // IsOnCurve checks if a point is on the twisted Edwards curve
 func (p *PointAffine) IsOnCurve() bool {
 
-	ecurve := GetEdwardsCurve()
-
```

Comment:
The problem here is that `GetEdwardsCurve` forces the construction to allocate a `big.Int`, which escapes to the heap.
But, most of the callers aren't really interested in the `order` field, but in immutable fields of the curve.

This was the top 1 source of allocs.